src/components/tabController/TabBar.tsx

```diff
@@ -357,44 +309,54 @@ const TabBar = (props: Props) => {
       <Reanimated.View style={[styles.selectedIndicator, indicatorStyle, _indicatorTransitionStyle]}/>
     ) : undefined;
 
-  const renderCodeBlock = () => {
+  const renderCodeBlock = _.memoize(() => {
     const nodes: any[] = [];
 
     nodes.push(set(_indicatorOffset,
       interpolate(currentPage, {
-        inputRange: itemsOffsets.map((_v, i) => i),
-        outputRange: itemsOffsets
+        inputRange: indicatorOffsets.map((_v, i) => i),
+        outputRange: indicatorOffsets
       })));
     nodes.push(set(_indicatorWidth,
-      interpolate(currentPage, {inputRange: itemsWidths.map((_v, i) => i), outputRange: itemsWidths})));
+      interpolate(currentPage, {
+        inputRange: itemsWidths.map((_v, i) => i),
+        outputRange: itemsWidths.map(v => v - 2 * INDICATOR_INSET)
+      })));
 
-    nodes.push(Reanimated.onChange(targetPage, Reanimated.call([targetPage], focusSelected)));
+    nodes.push(Reanimated.onChange(targetPage, Reanimated.call([targetPage], focusIndex as any)));
 
-    return <Code>{() => block(nodes)}</Code>;
-  };
+    const temp = <Code>{() => block(nodes)}</Code>;
+    return temp;
+  });
+
+  const shadowStyle = useMemo(() => {
+    return enableShadow ? propsShadowStyle || styles.containerShadow : undefined;
+  }, [enableShadow, propsShadowStyle]);
+
+  const _containerStyle = useMemo(() => {
+    return [styles.container, shadowStyle, {width: containerWidth}, containerStyle];
+  }, [shadowStyle, containerWidth, containerStyle]);
+
+  const indicatorContainerStyle = useMemo(() => {
+    return [styles.tabBar, !_.isUndefined(height) && {height}, {backgroundColor}];
+  }, [height, backgroundColor]);
+
+  const scrollViewContainerStyle = useMemo(() => {
+    return {minWidth: containerWidth};
+  }, [containerWidth]);
 
-  const shadowStyle = enableShadow ? propsShadowStyle || styles.containerShadow : undefined;
   return (
-    <View style={[styles.container, shadowStyle, {width: containerWidth}, containerStyle]}>
+    <View style={_containerStyle}>
       <FadedScrollView
-        // @ts-ignore TODO: typescript
+        /**
+         // @ts-ignore TODO: typescript */
         ref={tabBar}
         horizontal
-        contentContainerStyle={{minWidth: containerWidth}}
-        scrollEnabled={scrollEnabled}
-        onScroll={onScroll}
+        contentContainerStyle={scrollViewContainerStyle}
+        scrollEnabled // TODO:
         testID={testID}
-        snapToOffsets={snapBreakpoints}
       >
-        <View
-          style={[
-            styles.tabBar,
-            !_.isUndefined(height) && {height},
-            {paddingHorizontal: centerOffset, backgroundColor}
-          ]}
-        >
-          {renderTabBarItems}
-        </View>
+        <View style={indicatorContainerStyle}>{renderTabBarItems}</View>
         {renderSelectedIndicator}
```

Comment:
this should be `selectedIndicator` since it's an element and not a function

Reply:
Done